src/tokenizer.rs

```diff
@@ -510,7 +479,22 @@ impl<'a> Tokenizer<'a> {
                     }
                 }
                 '=' => self.consume_and_return(chars, Token::Eq),
-                '.' => self.consume_and_return(chars, Token::Period),
+                '.' => {
+                    chars.next(); // consume '.'
+                    match chars.peek() {
+                        Some('0'..='9') => {
+                            // Add the '.' back to the chars and parse as number.
+                            let mut chars_w_leading_zero = ".".to_string();
+                            while let Some(token) = chars.next() {
+                                chars_w_leading_zero.push(token);
+                            }
+                            let mut peekable = chars_w_leading_zero.chars().peekable();
+
+                            self.tokenize_number(&mut peekable)
```

Comment:
Could be wrong—haven't actually tried—but I think this mishandles something like `..07`. I'd recommend refactoring `tokenize_number` to take `seen_decimal` as a bool parameter, and then initialize `s` inside of `tokenize_number` to `.` if `seen_decimal` is true.